File: javascript/8kyu/sum-of-positive.js
// You get an array of numbers, return the sum of all of the positives ones.

// My Solution

function positiveSum(arr) {
  return arr
    .filter((n) => n > 0)
    .reduce((acc, n) => acc + n, 0);
}

// In Single Line Arrow Function Expression

const positiveSum = (arr) => arr.filter((n) => n > 0).reduce((acc, n) => acc + n, 0);

// Without Array Methods

function positiveSum(arr) {
  var total = 0;
  for (i = 0; i < arr.length; i++) {
    if (arr[i] > 0) {
      total += arr[i];
    }
  }
  return total;
}

// Reduce & Ternary

function positiveSum(arr) {
  return arr.reduce((a,b)=> a + (b > 0 ? b : 0),0);
}

File: python/8kyu/sum-of-positive.py
# You get an array of numbers, return the sum of all of the positives ones.
# Example [1,-4,7,12] => 1 + 7 + 12 = 20
# Note: if there is nothing to sum, the sum is default to 0.

# My Solution
def positive_sum(arr):
    result = 0
    for n in arr:
        if n > 0:
            result += n
    return result

# Another Solution
def positive_sum(arr):
    return sum(x for x in arr if x > 0)